src/kudu/consensus/leader_election.h

```diff
@@ -97,37 +120,131 @@ class VoteCounter {
 
 class FlexibleVoteCounter : public VoteCounter {
  public:
-  FlexibleVoteCounter(const LastKnownLeader& last_known_leader, RaftConfigPB config);
+  FlexibleVoteCounter(
+      const int64_t election_term, const LastKnownLeader& last_known_leader,
+      RaftConfigPB config);
 
   // Synchronization is done by the LeaderElection class. Therefore, VoteCounter
   // class doesn't need to take care of thread safety of its book-keeping
   // variables.
   Status RegisterVote(
-      const std::string& voter_uuid, ElectionVote vote,
+      const std::string& voter_uuid, VoteInfo vote,
```

Comment:
Same comment about vote_info as in the base class
```suggestion
      const std::string& voter_uuid, const VoteInfo& vote,
```